modules/core/packet-server/keeper/keeper.go

```diff
@@ -19,3 +31,82 @@ func NewKeeper(cdc codec.BinaryCodec, channelKeeper types.ChannelKeeper, clientK
 		ClientKeeper:  clientKeeper,
 	}
 }
+
+// Logger returns a module-specific logger.
+func (Keeper) Logger(ctx sdk.Context) log.Logger {
+	// TODO: prefix some submodule identifier?
+	return ctx.Logger().With("module", "x/"+exported.ModuleName)
+}
+
+func (k Keeper) RecvPacket(
+	ctx sdk.Context,
+	_ *capabilitytypes.Capability,
+	packet channeltypes.Packet,
```

Comment:
Should we call packet.ValidateBasic here? It doesn't seem like we did that originally, but could be useful since counterparty may be implemented incorrectly